### PR TITLE
enhance(webhook): handle repository transfer events

### PR DESCRIPTION
**Pull request by ecrupper**
Closes https://github.com/go-vela/community/issues/683

Also fixed a bug where the secrets weren't actually getting transferred because we were searching for secrets belonging to the new repo name, rather than the old. 

Also added a util function to split the full name of a repo to its org and repo respectively since that was used twice.

### CODE REVIEW

**Review by KellyMerrick — APPROVED** [review]
lgtm

File: scm/github/testdata/hooks/repository_transferred.json [hunk 1]
```diff
@@ -0,0 +1,159 @@
+{
+    "action": "transferred",
+    "changes": {
+        "owner": {
+            "from": {
+                "user": {
+                    "login": "Old-Codertocat",
```
Comment (plyr4) [inline scm/github/testdata/hooks/repository_transferred.json:7]: are these supposed to be octocat?

File: api/webhook/post.go [hunk 2]
```diff
@@ -714,7 +714,7 @@ func handleRepositoryEvent(c *gin.Context, m *types.Metadata, h *library.Hook, r
 
 	switch h.GetEventAction() {
 	// if action is rename, go through rename routine
-	case constants.ActionRenamed:
+	case constants.ActionRenamed, "transferred":
```
Comment (plyr4) [inline api/webhook/post.go:717]: should we make it a constant?

File: api/webhook/post.go [hunk 3]
```diff
@@ -779,34 +779,16 @@ func handleRepositoryEvent(c *gin.Context, m *types.Metadata, h *library.Hook, r
 // renameRepository is a helper function that takes the old name of the repo,
 // queries the database for the repo that matches that name and org, and updates
 // that repo to its new name in order to preserve it. It also updates the secrets
-// associated with that repo.
+// associated with that repo as well as build links for the UI.
 func renameRepository(h *library.Hook, r *library.Repo, c *gin.Context, m *types.Metadata) (*library.Repo, error) {
 	logrus.Debugf("renaming repository from %s to %s", r.GetPreviousName(), r.GetName())
```
Comment (wass3rw3rk) [inline api/webhook/post.go:784]: what do you think about increasing log level here? feels like something that should be in audit and not only visibly in debug mode.

**Review by wass3rw3rk — APPROVED** [review]

**Review by timhuynh94 — APPROVED** [review]

**Review by wass3rw3rk — APPROVED** [review]

**Pull request merged**